clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v2/artifact/ArtifactReplacer.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright 2017 Google, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.netflix.spinnaker.clouddriver.kubernetes.v2.artifact;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.jayway.jsonpath.Configuration;
+import com.jayway.jsonpath.DocumentContext;
+import com.jayway.jsonpath.JsonPath;
+import com.jayway.jsonpath.spi.json.JacksonJsonNodeJsonProvider;
+import com.jayway.jsonpath.spi.mapper.JacksonMappingProvider;
+import com.netflix.spinnaker.clouddriver.kubernetes.v2.description.manifest.KubernetesManifest;
+import com.netflix.spinnaker.kork.artifacts.model.Artifact;
+import lombok.Builder;
+import lombok.extern.slf4j.Slf4j;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+@Slf4j
+public class ArtifactReplacer {
+  private static final ObjectMapper mapper = new ObjectMapper();
+  private static final Configuration configuration = Configuration.builder()
+      .jsonProvider(new JacksonJsonNodeJsonProvider())
+      .mappingProvider(new JacksonMappingProvider())
+      .build();
+
+  List<Replacer> replacers = new ArrayList<>();
+
+  public ArtifactReplacer addReplacer(Replacer replacer) {
+    replacers.add(replacer);
+    return this;
+  }
+
+  public KubernetesManifest replaceAll(KubernetesManifest input, List<Artifact> artifacts) {
+    DocumentContext document;
+    try {
+      document = JsonPath.using(configuration).parse(mapper.writeValueAsString(input));
+    } catch (JsonProcessingException e) {
+      log.error("Malformed manifest - not sure how this is possible", e);
```

Comment:
> not sure how this is possible

Remove this and replace with an explanation of why this is impossible.

Reply:
An explanation is overkill - should be clear from the line above (reading json as json). But I'll drop it from the message.